proto/buf/alpha/registry/v1alpha1/doc.proto

```diff
@@ -377,6 +387,40 @@ message FieldOptions {
   int32 jstype = 4;
 }
 
+// Option provides information for a descriptor option that can be show in documentations.
+message Option {
+  // The name of the option.
+  string name = 1;
+  // The value of an option.
+  LiteralValue value = 2;
+}
+
+// MessageLiteral represents an option value for a type that is a message.
+message MessageLiteral {
+  map<string, LiteralValue> fields = 1;
+}
+
+// ArrayLiteral represents an option value for a type that is a repeated field.
+// Maps are also represented as ArrayLiterals where each elements are a MessageLiteral
+// that contains a key and value field.
+message ArrayLiteral {
+  repeated LiteralValue elements = 1;
+}
+
+// LiteralValue represents the actual value of the extension.
+message LiteralValue {
+  oneof value {
+    string string_value = 2;
+    int64 int_value = 3;
+    uint64 uint_value = 4;
+    double double_value = 5;
+    bool bool_value = 6;
+    string enum_value = 7;
+    ArrayLiteral array_value = 8;
+    MessageLiteral message_value = 9;
+  }
+}
```

Comment:
I'm not sure about our stance on nesting messages. But, just to not add to the already broad `buf.alpha.registry.v1alpha1` namespace, consider nesting these inside of `DescriptorOption`.